queries.sql

```diff
@@ -15,3 +15,139 @@ SELECT * FROM animals WHERE neutered = True;
+
+SELECT O.full_name AS OWNER, A.name AS ANIMAL_NAMES FROM owners O LEFT JOIN animals A ON O.id = A.owner_id;
+
+-- How many animals are there per species ?
+
+SELECT S.name AS SPECIES_TYPE, COUNT(A.name) AS NUMBER_PER_SPECIES FROM species S JOIN animals A ON S.id = A.species_id GROUP BY S.name;
+
+-- List all Digimon owned by Jennifer Orwell
+
+SELECT O.full_name AS OWNER, A.name AS ANIMAL_NAMES, S.name AS SPECIES_TYPE FROM owners O JOIN animals A ON O.id = A.owner_id 
+JOIN species S ON S.id = A.species_id WHERE O.full_name = 'Jennifer Orwell' AND S.name = 'Digimon';
+
+-- List all animals owned by Dean Winchester that haven't tried to escape
+
+SELECT owners.full_name AS OWNER, animals.name AS ANIMAL_NAMES, animals.escape_attempts AS ESCAPE_ATTEMPTS FROM animals
+JOIN owners ON owners.id = animals.owner_id WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
+
+-- Who owns the most animals ?
+
+SELECT O.full_name AS OWNER, COUNT(a.name) AS TOTAL_ANIMALS FROM owners AS O LEFT JOIN animals AS A
+ON A.owner_id = O.id GROUP BY O.full_name ORDER BY COUNT(A.name) DESC LIMIT 1;
+
+-- Write queries to answer the following:
+
+-- Who was the last animal seen by William Tatcher ?
+
+SELECT vets.name AS vets_name, animals.name AS animal_name, visits.date_of_visit AS date_of_visit FROM vets
+INNER JOIN visits ON vets.id = visits.vet_id INNER JOIN animals ON animals.id = visits.animal_id WHERE vets.name = 'William Tatcher'
+ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see ?
+
+SELECT vets.name AS vets_name, COUNT(visits.animal_id) AS seen_animals FROM vets INNER JOIN visits 
+ON vets.id = visits.vet_id WHERE vets.name = 'Stephanie Mendez' GROUP BY vets.name;
+
+-- List all vets and their spcialties, including vets with no specialties.
+
+SELECT vets.name AS vets_name, species.name AS specialties FROM vets LEFT JOIN specializations ON vets.id = specializations.vet_id
+LEFT JOIN species ON specializations.species_id = species.id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+
+SELECT vets.name AS vets_name, animals.name AS animal_name, visits.date_of_visit AS date_of_visit FROM vets
+INNER JOIN visits ON vets.id = visits.vet_id INNER JOIN animals ON visits.animal_id = animals.id WHERE 
+vets.name = 'Stephanie Mendez' AND visits.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
+
+-- What animal has the most visits to vets?
+
+SELECT animals.name AS animal_name, COUNT(animals.id) AS most_visits FROM visits
+INNER JOIN animals ON visits.animal_id = animals.id GROUP BY animals.name ORDER BY COUNT(animals.id) DESC LIMIT 1;
+
+-- Who was Maisy Smith's first visit?
+
+SELECT vets.name AS vets_name, visits.date_of_visit AS date_of_visit FROM vets
+INNER JOIN visits ON vets.id = visits.vet_id WHERE vets.name = 'Maisy Smith' ORDER BY visits.date_of_visit
+LIMIT 1;
```

Comment:
- [ ] Please kindly include the animal name in the select column as the query asks `who`

Reply:
Well resolved 👍🏽